### PR TITLE
Run Linux tests in Docker

### PR DESCRIPTION
**Pull request by vasi**
Tests can be run in docker with `./test/run_tests.sh`. And these are run in CI!

Allows users/devs to run tests in a fully reproducible environment. This means we shouldn't be affected by things like changes in libc's `rand()`, any changes Github makes to their Actions runners, etc. Can even run Linux tests accurately in macOS.

### CODE REVIEW

**Comment by vasi** [comment]
(Note I marked this as for merging into master, even though it includes #12 . Unfortunately GH has weird behavior when you create a PR for merging into a feature branch. We can fixup the merge target later, depending what we decide to do with this.)

**Comment by vasi** [comment]
Ok, made CI use Docker as well, so we have local+CI reproducibility.

**Comment by bhearsum** [comment]
I think this is a wonderful approach, particularly because it has us using the same image locally and in CI. (Too bad we can't also run the reference tests in Docker so we could run them on Linux...)

I'll hold off formalling reviewing it until #12 is merged and this is rebased, but everything here looks fine to me.

**Comment by vasi** [comment]
This should be ready now. @bhearsum could you kick off CI on this PR?

**Comment by vasi** [comment]
CI seems to pass here: https://github.com/vasi/libdmg-hfsplus/actions/runs/10906688243/job/30268515760

**Comment by vasi** [comment]
Runs great on macOS using Docker Desktop, too.

**Review by bhearsum — APPROVED** [review]
This is fantastic, thank you!

**Pull request merged**
